This patch is based on the upstream commit described below, adapter for use
in the Debian package by Peter Michael Green.

commit 68531230aa061034d6144adf19ebc681bb8d5253
Author: Neal H. Walfield <neal@pep.foundation>
Date:   Fri Feb 10 11:28:50 2023 +0100

    Update to rpassword 6.0.
    
      - Debian has moved to rpassword 6.0, which has a different API from
        rpassword 5.0.  Follow Debian.
    
      - Fixes #18.

Index: sequoia-keyring-linter/src/keyring-linter.rs
===================================================================
--- sequoia-keyring-linter.orig/src/keyring-linter.rs
+++ sequoia-keyring-linter/src/keyring-linter.rs
@@ -54,11 +54,11 @@ fn decrypt_key<R>(key: Key<key::SecretPa
                 let mut first = true;
                 loop {
                     // Prompt the user.
-                    match rpassword::read_password_from_tty(
-                        Some(&format!(
+                    match rpassword::prompt_password(
+                        &format!(
                             "{}Enter password to unlock {} (blank to skip): ",
                             if first { "" } else { "Invalid password. " },
-                            key.keyid().to_hex())))
+                            key.keyid().to_hex()))
                     {
                         Ok(p) => {
                             first = false;
Index: sequoia-keyring-linter/Cargo.toml
===================================================================
--- sequoia-keyring-linter.orig/Cargo.toml
+++ sequoia-keyring-linter/Cargo.toml
@@ -45,7 +45,7 @@ version = "1"
 version = "0.2"
 
 [dependencies.rpassword]
-version = "5.0"
+version = "6.0"
 
 [dependencies.sequoia-openpgp]
 version = "1.0"
